sticker-utils/src/subcommands/annotate.rs

```diff
@@ -0,0 +1,211 @@
+use std::fs::File;
+use std::io::BufWriter;
+
+use clap::{App, Arg, ArgMatches};
+use conllx::io::{ReadSentence, Reader, WriteSentence, Writer};
+use stdinout::{Input, OrExit, Output};
+use sticker::config::{Config, TomlRead};
+use sticker::encoders::Encoders;
+use sticker::input::WordPieceTokenizer;
+use sticker::model::BertModel;
+use sticker::tagger::Tagger;
+use tch::nn::VarStore;
+use tch::{self, Device};
+
+use crate::progress::TaggerSpeed;
+use crate::sent_proc::SentProcessor;
+use crate::traits::{StickerApp, DEFAULT_CLAP_SETTINGS};
+
+const BATCH_SIZE: &str = "BATCH_SIZE";
+const CONFIG: &str = "CONFIG";
+const GPU: &str = "GPU";
+const INPUT: &str = "INPUT";
+const MAX_LEN: &str = "MAX_LEN";
+const OUTPUT: &str = "OUTPUT";
+const READ_AHEAD: &str = "READ_AHEAD";
+
+pub struct AnnotateApp {
+    batch_size: usize,
+    config: String,
+    device: Device,
+    input: Option<String>,
+    max_len: Option<usize>,
+    output: Option<String>,
+    read_ahead: usize,
+}
+
+impl AnnotateApp {
+    fn process<R, W>(&self, tokenizer: &WordPieceTokenizer, tagger: Tagger, read: R, write: W)
+    where
+        R: ReadSentence,
+        W: WriteSentence,
+    {
+        let mut speed = TaggerSpeed::new();
+
+        let mut sent_proc = SentProcessor::new(
+            &tokenizer,
+            &tagger,
+            write,
+            self.batch_size,
+            self.max_len,
+            self.read_ahead,
+        );
+
+        for sentence in read.sentences() {
+            let sentence = sentence.or_exit("Cannot parse sentence", 1);
+            sent_proc
+                .process(sentence)
+                .or_exit("Error processing sentence", 1);
+
+            speed.count_sentence()
+        }
+    }
+}
+
+impl StickerApp for AnnotateApp {
+    fn app() -> App<'static, 'static> {
+        App::new("annotate")
+            .settings(DEFAULT_CLAP_SETTINGS)
+            .about("Annotate a corpus")
+            .arg(
+                Arg::with_name(CONFIG)
+                    .help("Sticker configuration file")
+                    .index(1)
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name(INPUT)
+                    .help("Input data")
+                    .long("input")
+                    .takes_value(true),
+            )
+            .arg(
+                Arg::with_name(OUTPUT)
+                    .help("Output data")
+                    .long("output")
+                    .takes_value(true),
+            )
+            .arg(
+                Arg::with_name(BATCH_SIZE)
+                    .long("batch-size")
+                    .takes_value(true)
+                    .help("Batch size")
+                    .default_value("32"),
+            )
+            .arg(
```

Comment:
Do we also want to give the option to explicitly use CPU instead of relying on `CUDA_VISIBLE_DEVICES`/`HIP_VISIBLE_DEVICES`? (or is this already happening when not giving a value to --gpu?)

Reply:
nevermind, saw it further down.